api-reference/_hidden/GridBaseColumn/calculateCellValue.md

```diff
@@ -363,12 +361,12 @@ To invoke the default behavior, call the **this.defaultCalculateCellValue(rowDat
 
 ---
 
-[/note]
-
 #include uiwidgets-ref-functioncontext with { 
     value: "column's configuration"
 }
 
+[important] Do not implement complex calculations within this method to avoid poor widget's performance. This method is called every time a row renders or when a sorting/filtering/summary setting changes. Thus, it can be called multiple times in a short period and multiple complex calculations can slow down the widget.
```

Comment:
We warn the reader but do not provide a way to solve the issue. What calculation is considered complex? What to do if the reader still needs this calculation to be performed?

@vconst Can't we fix this in code? Why does this function need to be called when data is sorted, filtered, etc.? Why recalculate the values when they were already calculated?

Reply:
Adding cache for such cases is not simple task. And in most of user cases cache is not necessary and it may be bad because cache takes up addition memory.

If user has large calculations in caclulateCellValue, he can add caching to his own code.

So, customers should know that caclulateCellValue is called many times and they should optimize their own code by simplifing or by adding cache.

Reply:
> If user has large calculations in caclulateCellValue, he can add caching to his own code.

Any good articles about caching to which we could forward the reader?

Reply:
I do not have any specific links for this theme